graphqlmetrics/core/server.go

```diff
@@ -39,16 +44,47 @@ func NewServer(metricsService graphqlmetricsv1connect.GraphQLMetricsServiceHandl
 
 func (s *Server) bootstrap() {
 	mux := http.NewServeMux()
+
 	path, handler := graphqlmetricsv1connect.NewGraphQLMetricsServiceHandler(
 		s.metricsService,
 		// Brotli compression support.
 		brotli.WithCompression(),
 	)
-	mux.Handle("/health", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	healthHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
-	}))
+	})
+
+	if s.metricConfig.OpenTelemetry.Enabled {
+		tp, err := s.metricConfig.NewTracerProvider()
+		if err != nil {
+			s.logger.Error("Error creating tracing provider", zap.Error(err))
+		}
+		defer func() {
+			if err := tp.Shutdown(context.Background()); err != nil {
```

Comment:
Shutdown logic should be done in `Shutdown` with a proper context.

Reply:
Would it be ok to use the same context as here as the server itself?

Reply:
I moved the shutdown together, seems to work fine to reuse the context.

Let me know if this might be an issue.